wsd/COOLWSD.cpp

```diff
@@ -3575,7 +3577,6 @@ class PrisonerRequestDispatcher final : public WebSocketHandler
 #else
             pid_t pid = 100;
             std::string jailId = "jail";
-            LOG_ASSERT_MSG(socket->getInBuffer().empty(), "Unexpected data in prisoner socket");
```

Comment:
Note: on mobile, the socket is always empty as JavaScript messages come over a separate "fake socket" class.

Reply:
Great! So why is asserting that it *is* empty wrong?

Reply:
> Great! So why is asserting that it _is_ empty wrong?

Because that socket is *always* empty. This code is called only once when a document is first opened and I doubt that an actual C socket is ever created so the socket buffer will always be empty.

I would guess that the only reason that iOS invokes this code is that is a side-effect of bootstrapping LibreOffice and starting the LOKit queue. After this code is called, iOS immediately tells the JS code to set app.Socket to a fake socket. The Online server components are never sent messages directly on mobile.

Instead, on iOS the fake socket routes messages to -[DocumentViewController userContentController:didReceiveScriptMessage:] in ios/Mobile/DocumentViewController.mm. This code forwards the message directly to LOKit, bypassing the Online server completely.